rules/opentitan.bzl

```diff
@@ -726,12 +771,107 @@ pick_correct_archive_for_device = rv_rule(
     toolchains = ["@rules_cc//cc:toolchain_type"],
 )
 
+def opentitan_multislot_flash_binary(
+        name,
+        srcs,
+        image_size,
+        key_per_binary,
+        offset_per_binary,
+        platform = OPENTITAN_PLATFORM):
+    """A helper macro for generating multislot OpenTitan binary flash images.
+
+    This macro is mostly a wrapper around the `assemble_flash_image` rule, that
+    invokes `opentitantool` to stitch together multiple `opentitan_flash_binary`
+    images to create a single image for bootstrapping. This enables efficient
+    testing by only requiring one boostrap operation to load both silicon
+    creator and owner stages of flash.
+    Args:
+      @param name: The name of this rule.
+      @param srcs: A list of `opentitan_flash_binary` targets to stich together.
+      @param image_size: The final flash image_size to pass to `opentitantool`.
```

Comment:
Can you remind me why we want to customize this? What about setting this to the size of the flash by default?

If this was due to efficiency reasons, i.e. writing a ton of `0xff`s, I have a WIP PR that makes opentitantool skip all 0xff chunks during bootstrapping.

Reply:
I think originally it was efficiency reasons, and I wasn't sure if this was a negative test path we wanted to excercise? I can update this once your PR lands? or sooner if you prefer.

Reply:
I just tried to remove and hardcode the size of flash (`0x100000`). However, the flash VMEM scrambling rule (which invokes a script that does not do any scrambling currently but rather just computes ECC bits) takes a really long time (the build didn't finish after 10 min on my workstation) due to all the `0xff` bytes were filled, so I kept the image size parameter for now to avoid blowing up CI runtimes. I can remove this parameter in a follow up PR that optimizes the [`gen-flash-img.py`](https://cs.opensource.google/opentitan/opentitan/+/master:util/design/BUILD;drc=74f9095ab484c27197d232a23a01e10786df8392;l=19) script that is taking so long to run.

Reply:
Oof! Thanks for the update.